### PR TITLE
Update model and UI with final columns

### LINKED ISSUE #12: 7 - Adjust GUI to accept the most effective columns

### PR DESCRIPTION
**Pull request by robotbread2**
Changed models.py to only train on base_egg_steps, base_total and capture_rate, and removed some preprocessing that was no longer needed. The GUI will now request for base_egg_steps, capture_rate and the 6 columns base_total is composed of. 

A few new changes were also made to the GUI:

> Added window scaling
> Added exception handling for bad input
> Columns left blank will now use an average instead of zero.

Running models.py should now produce pickles which are compatible with the app.

Resolves #12

### CODE REVIEW

File: deployed_model/app.py [hunk 1]
```diff
@@ -58,82 +57,89 @@ def unpickle(model_file : str):
...
     except Exception as e:
         print(f"Input error: {e}")
-        return None
+        error_df = pd.DataFrame()
+        return error_df
 
 # Given a pandas dataframe in the a format used for training, predict if the pokemon is legendary
 
 def predict_is_legendary(df, model) -> bool:
     result = model.predict(df)
```
Comment (Endermandels) [inline deployed_model/app.py:141]: I get this error every time i press predict:

Traceback (most recent call last):
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 305, in <module>
    main()
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 302, in main
    run_app(model, xs)
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 267, in run_app
    is_legendary = predict_is_legendary(test_df, model)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 141, in predict_is_legendary
    result = model.predict(df)
             ^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\sklearn\pipeline.py", line 600, in predict       
    Xt = transform.transform(Xt)
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\sklearn\utils\_set_output.py", line 316, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\app.py", line 26, in transform
    return xs[self.columns]
           ~~^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\frame.py", line 4108, in __getitem__ 
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\indexes\base.py", line 6200, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "C:\Users\elija\Desktop\Python Stuff\YAML-G\deployed_model\virt\Lib\site-packages\pandas\core\indexes\base.py", line 6252, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['against_bug', 'against_dark', 'against_dragon', 'against_electric', 'against_fairy', 'against_fight', 'against_fire', 'against_flying', 'against_ghost', 'against_grass', 'against_ground', 'against_ice', 'against_normal', 'against_poison', 'against_psychic', 'against_rock', 'against_steel', 'against_water', 'attack', 'base_happiness', 'defense', 'experience_growth', 'height_m', 'hp', 'percentage_male', 'pokedex_number', 'sp_attack', 'sp_defense', 'speed', 'weight_kg', 'type1_dark', 'type1_dragon', 'type1_electric', 'type1_fairy', 'type1_fighting', 'type1_fire', 'type1_flying', 'type1_ghost', 'type1_grass', 'type1_ground', 'type1_ice', 'type1_normal', 'type1_poison', 'type1_psychic', 'type1_rock', 'type1_steel', 'type1_water', 'type2_bug', 'type2_dark', 'type2_dragon', 'type2_electric', 'type2_fairy', 'type2_fighting', 'type2_fire', 'type2_flying', 'type2_ghost', 'type2_grass', 'type2_ground', 'type2_ice', 'type2_normal', 'type2_poison', 'type2_psychic', 'type2_rock', 'type2_steel', 'type2_water', 'Adaptability', 'Aftermath', 'Air Lock', 'Analytic', 'Anger Point', 'Anticipation', 'Arena Trap', 'Aroma Veil', 'Bad Dreams', 'Battery', 'Battle Armor', 'Battle Bond', 'Beast Boost', 'Berserk', 'Big Pecks', 'Blaze', 'Bulletproof', 'Cheek Pouch', 'Chlorophyll', 'Clear Body', 'Cloud Nine', 'Color Change', 'Comatose', 'Competitive', 'Compoundeyes', 'Contrary', 'Corrosion', 'Cursed Body', 'Cute Charm', 'Damp', 'Dancer', 'Dark Aura', 'Dazzling', 'Defeatist', 'Defiant', 'Download', 'Drizzle', 'Drought', 'Dry Skin', 'Early Bird', 'Effect Spore', 'Emergency Exit', 'Fairy Aura', 'Filter', 'Flame Body', 'Flare Boost', 'Flash Fire', 'Flower Gift', 'Flower Veil', 'Fluffy', 'Forewarn', 'Friend Guard', 'Frisk', 'Full Metal Body', 'Fur Coat', 'Gale Wings', 'Galvanize', 'Gluttony', 'Gooey', 'Grass Pelt', 'Grassy Surge', 'Guts', 'Harvest', 'Healer', 'Heatproof', 'Heavy Metal', 'Honey Gather', 'Huge Power', 'Hustle', 'Hydration', 'Hyper Cutter', 'Ice Body', 'Illuminate', 'Illusion', 'Immunity', 'Imposter', 'Infiltrator', 'Innards Out', 'Inner Focus', 'Insomnia', 'Intimidate', 'Iron Barbs', 'Iron Fist', 'Justified', 'Keen Eye', 'Klutz', 'Leaf Guard', 'Levitate', 'Light Metal', 'Lightningrod', 'Limber', 'Liquid Ooze', 'Liquid Voice', 'Long Reach', 'Magic Bounce', 'Magic Guard', 'Magician', 'Magma Armor', 'Magnet Pull', 'Marvel Scale', 'Mega Launcher', 'Merciless', 'Minus', 'Misty Surge', 'Mold Breaker', 'Moody', 'Motor Drive', 'Moxie', 'Multiscale', 'Mummy', 'Natural Cure', 'No Guard', 'Normalize', 'Oblivious', 'Overcoat', 'Overgrow', 'Own Tempo', 'Pickpocket', 'Pickup', 'Pixilate', 'Plus', 
'Poison Heal', 'Poison Point', 'Poison Touch', 'Power of Alchemy', 'Prankster', 'Pressure', 'Prism Armor', 'Protean', 'Psychic Surge', 'Pure Power', 'Queenly Majesty', 'Quick Feet', 'RKS System', 'Rain Dish', 'Rattled', 'Receiver', 'Reckless', 'Refrigerate', 'Regenerator', 'Rivalry', 'Rock Head', 'Rough Skin', 'Run Away', 'Sand Force', 'Sand Rush', 'Sand Stream', 'Sand Veil', 'Sap Sipper', 'Schooling', 'Scrappy', 'Serene Grace', 'Shadow Shield', 'Shadow Tag', 'Shed Skin', 'Sheer Force', 'Shell Armor', 'Shield Dust', 'Shields Down', 'Simple', 'Skill Link', 'Slow Start', 'Slush Rush', 'Sniper', 'Snow Cloak', 'Snow Warning', 'Solar Power', 'Solid Rock', 'Soul-Heart', 'Soundproof', 'Speed Boost', 'Stakeout', 'Stall', 'Stamina', 'Stance Change', 'Static', 'Steadfast', 'Steelworker', 'Stench', 'Sticky Hold', 'Storm Drain', 'Strong Jaw', 'Sturdy', 'Suction Cups', 'Super Luck', 'Surge Surfer', 'Swarm', 'Sweet Veil', 'Swift Swim', 'Symbiosis', 'Synchronize', 'Tangled Feet', 'Tangling Hair', 'Technician', 'Telepathy', 'Teravolt', 'Thick Fat', 'Tinted Lens', 'Torrent', 'Tough Claws', 'Toxic Boost', 'Trace', 'Triage', 'Truant', 'Turboblaze', 'Unaware', 'Unburden', 'Unnerve', 'Victory Star', 'Vital Spirit', 'Volt Absorb', 'Water Absorb', 'Water Bubble', 'Water Compaction', 'Water Veil', 'Weak Armor', 'White Smoke', 'Wonder Guard', 'Wonder Skin ', 'Zen Mode'] not in index"

**Review by Endermandels — CHANGES_REQUESTED** [review]
The buttons look great, but the predict button causes the app to crash.

File: deployed_model/app.py (hunk 1 above)
Comment (robotbread2) [reply deployed_model/app.py:141]: It looks like its trying to use a bunch of the older columns - try running models.py again (the one in this branch) to make new pickles and give it another go.

**Review by Endermandels — APPROVED** [review]
Everything works after running models.py again.

**Pull request merged**